### ISSUE TITLE
ipopt solver not installed

### ISSUE THREAD

**Issue by AbdulRafay** [PyPSA/pypsa-eur#710]
I am using default master of pypsa-eur. Created env using conda. I am trying to solve using Ipopt solver which i installed from conda-forge channel

`conda install -y -c conda-forge ipopt `

I am trying to run tutorial: Electricity- Only, my config/test/config.electricity.yaml looks like following:

```
solving:
  solver:
    name: ipopt
    options: {}

```

when I execute:

`snakemake -call results/test-elec/networks/elec_s_6_ec_lcopt_Co2L-24H.nc --configfile config/test/config.electricity.yaml`

it gives me following error:

```
Config file config/config.yaml is extended by additional config specified via the command line.
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 2
Rules claiming more threads will be scaled down.
Provided resources: mem_mb=1396, mem_mib=1332
Conda environments: ignored
Select jobs to execute...
Changing to shadow directory: /mnt/d/inc/pypsa_test/pypsa-eur/.snakemake/shadow/tmp749heia3
INFO:pypsa.io:Imported network elec_s_6_ec_lcopt_Co2L-24H.nc has buses, carriers, generators, global_constraints, lines, links, loads, storage_units, stores
INFO:linopy.model: Solve problem using Ipopt solver
Traceback (most recent call last):
  File "/mnt/d/inc/pypsa_test/pypsa-eur/.snakemake/shadow/tmp749heia3/.snakemake/scripts/tmpemiv8m9t.solve_network.py", line 691, in <module>
    n = solve_network(
  File "/mnt/d/inc/pypsa_test/pypsa-eur/.snakemake/shadow/tmp749heia3/.snakemake/scripts/tmpemiv8m9t.solve_network.py", line 621, in solve_network
    status, condition = n.optimize(
  File "/home/john/anaconda3/envs/pypsa-eur/lib/python3.10/site-packages/pypsa/optimization/optimize.py", line 587, in __call__
    return optimize(self._parent, *args, **kwargs)
  File "/home/john/anaconda3/envs/pypsa-eur/lib/python3.10/site-packages/pypsa/optimization/optimize.py", line 567, in optimize
    status, condition = m.solve(solver_name=solver_name, **solver_options, **kwargs)
  File "/home/john/anaconda3/envs/pypsa-eur/lib/python3.10/site-packages/linopy/model.py", line 1015, in solve
    assert solver_name in available_solvers, f"Solver {solver_name} not installed"
AssertionError: Solver ipopt not installed
Not cleaning up /mnt/d/inc/pypsa_test/pypsa-eur/.snakemake/shadow/tmp749heia3/.snakemake/scripts/tmpemiv8m9t.solve_network.py
[Tue Aug  1 16:21:01 2023]
Error in rule solve_network:
    jobid: 0
    input: resources/networks/elec_s_6_ec_lcopt_Co2L-24H.nc
    output: results/test-elec/networks/elec_s_6_ec_lcopt_Co2L-24H.nc
    log: logs/test-elec/solve_network/elec_s_6_ec_lcopt_Co2L-24H_solver.log, logs/test-elec/solve_network/elec_s_6_ec_lcopt_Co2L-24H_python.log (check log file(s) for error details)
    conda-env: /mnt/d/inc/pypsa_test/pypsa-eur/.snakemake/conda/83197cba8a1d5f9e2cd7f1cba7fda0cb_

Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2023-08-01T162048.210318.snakemake.log
```

ipopt -v gives me `Ipopt 3.14.12 (Linux x86_64), ASL(20160307)`

**Comment by fneum** [comment]
@AbdulRafay, `conda list` when the PyPSA-Eur environment is activated should include `ipopt` in the list.

**Comment by AbdulRafay** [comment]
Even if I build ipopt from source and include it the PATH it gives me error `Solver ipopt not installed` in solve_network rule...

**Comment by FabianHofmann** [comment]
@AbdulRafay you actually found a misleading error message in linopy. The thing is, `ipopt` is not supported (and not `not installed`) as a solver here. Could you try another one?

**Comment by AbdulRafay** [comment]
> @AbdulRafay you actually found a misleading error message in linopy. The thing is, `ipopt` is not supported (and not `not installed`) as a solver here. Could you try another one?

I tried `HiGHS` and it worked well. One thing i noticed when using `CBC` was that at cluster network step `ipopt` was used to solved the quadratic equation and then CBC was used to solve network. But when i used ipopt for solving network it errored out at the last step. Cluster network step was executed fine even when using `ipopt` as a solver. 

Anyhow I ended up using Groubi but I hope this helps anyone in the future.
